### DISCUSSION TITLE
How do I add customer pages to Elsa Studio 3.0?

### DISCUSSION THREAD

**Discussion by dupies** [elsa-workflows/elsa-core#5931] [Q&A] [answered]
I'm using Elsa Studio Blazor Server, I would like to add customer pages to the application and extend or overwrite the Nav, 
is this possible? 

Tried looking around, didn't find anything that could help.

Thank you.

**Comment by dupies** [comment]
![image](https://github.com/user-attachments/assets/add69968-d514-4171-94a5-475718863c23)

Found this while going through the project. 

Is there a way to extend/overwrite this in my Blazor Server application and add custom menu items that route to custom pages that I would create?

My only other option is to fork the project and make the changes, I'd hat to do that because I'd have to maintain it on my own, It would be a lot more work getting updates.

**Comment by MeloHenrique** [comment, marked as answer]
I couldn't find any documentation on this, but it shouldn't be too difficult.

First, you need to add a _**Feature**_ class to the project where your pages are located. This class is essential because, without it, their custom App.razor won't load your assembly that contains the pages.

```
public class Feature : FeatureBase
{
    
} 
```

Next, create a _**Menu**_ class, which will allow you to add an additional item to the navigation menu.

```
public class TestMenu : IMenuProvider
{
    public ValueTask<IEnumerable<MenuItem>> GetMenuItemsAsync(CancellationToken cancellationToken = default)
    {
        var menuItems = new List<MenuItem>
        {
            new()
            {
                Icon = Icons.Material.Outlined.Lightbulb,
                Href = "test",
                Text = "Test",
                GroupName = MenuItemGroups.General.Name,
                Match = NavLinkMatch.All
            }
        };

        return new ValueTask<IEnumerable<MenuItem>>(menuItems);
    }
}
```

After that, you need to register both the _Feature_ and _TestMenu_ classes as scoped services in your _**Program.cs**_.

```
builder.Services.AddScoped<IFeature, Feature>();
builder.Services.AddScoped<IMenuProvider, TestMenu>();
```
Finally, add your page:

```
@page "/test"
<h3>Test</h3>

@code {
    
}
```

If there's anything else you need or if I missed something, feel free to reach out!

**Reply by dupies** [reply]
Thank you, worked!
